resources/environments/resource_id_broker_mappings.go

```diff
@@ -200,6 +201,21 @@ func isSetIDBEnvNotFoundError(err error) bool {
 	return false
 }
 
+func isEnvironmentStillExists(ctx context.Context, client *client.Environments, envName string, state *idBrokerMappingsResourceModel) bool {
+	envParams := operations.NewDescribeEnvironmentParamsWithContext(ctx)
+	envParams.WithInput(&environmentsmodels.DescribeEnvironmentRequest{
+		EnvironmentName: &envName,
+	})
+	_, err := client.Operations.DescribeEnvironment(envParams)
+	if err != nil {
```

Comment:
I don't think you should swallo the error if it is not environment not found. You should return it back from this function and let the caller error out.